src/Core/Framework/App/Lifecycle/Persister/ScriptPersister.php

```diff
@@ -105,10 +105,11 @@ public function refresh(): void
         $criteria->setTitle('app-scripts::refresh');
         $criteria->addFilter(new EqualsFilter('active', true));
 
-        $apps = $this->appRepository->search($criteria, Context::createDefaultContext())->getEntities();
+        /** @var array<string> $apps */
+        $apps = $this->appRepository->searchIds($criteria, Context::createDefaultContext())->getIds();
 
         foreach ($apps as $app) {
```

Comment:
Maybe rename the loop variable to be more concise.
```suggestion
        foreach ($appIds as $appId) {
```